pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -97,8 +99,9 @@ var _ reconcile.Reconciler = &ReconcileControlPlaneCerts{}
 // ReconcileControlPlaneCerts reconciles a ClusterDeployment object
 type ReconcileControlPlaneCerts struct {
 	client.Client
-	scheme  *runtime.Scheme
-	applier applier
+	scheme              *runtime.Scheme
+	applier             applier
+	remoteClientBuilder func(client.Client, *hivev1.ClusterDeployment) remoteclient.Builder
```

Comment:
My personal preference would be to remove the `client.Client` parameter from this function. That will match what is done in the other controllers. If you feel strongly about keeping it this way, then I can live with it, though,